### ISSUE TITLE
calculation mAP, ap=0.00%,precision=-nan(ind)

### ISSUE THREAD

**Issue by DCC-lzhy** [AlexeyAB/darknet#2539]
I just get one problem. I has change the file path as  it`s well done before. But I meet the problem this time, the context:
 calculation mAP (mean average precision)...
1052
 detections_count = 3, unique_truth_count = 1097
class_id = 0, name = vein,       ap = 0.00 %
 for thresh = 0.25, precision = -nan(ind), recall = 0.00, F1-score = -nan(ind)
 for thresh = 0.25, TP = 0, FP = 0, FN = 1097, average IoU = 0.00 %

 mean average precision (mAP) = 0.000000, or 0.00 %
Total Detection Time: 10.000000 Seconds

 mean_average_precision = 0.000000

**Comment by wonchulSon** [comment]
how did you get the map?

**Comment by DCC-lzhy** [comment]
just as the readme show, use the command: darknet.exe detector .data  .cfg -map .I change the network in the cfg, so I retrain it. It goes well before ,but when I optimize my dataset, it's map goes wrong, it's loss is decreasing normally. I alse tried use the command: darknet.exe detector map .data .cfg .weights. I also tried train unchanged yolov3-tiny on my optimized dataset, map is right as before. In the progress, one thing I did is optimizing my dataset by deleting the blurred pic.

**Comment by DCC-lzhy** [comment]
It's ok now.

**Comment by kenrubiooo** [comment]
> It's ok now.

Hi @DCC-lzhy,

How did you solve this problem? I am using the yolov3.weights, yolov3.cfg, and coco.names. All the APs are giving me 0.00%. I am just trying to get the mAP of the available weights files of yolov3 for 416x416 input.

**Comment by DCC-lzhy** [comment]
I have no idea about your problem, I meet it in the training process, and training loss is decreasing normall, but get 0 map in the validation, it just appeared aside one of my custom cfgs, I guess I used wrong learning rate strategy, so I keep the cfg same as the original, and it goes well.

**Comment by alphapibeta** [comment]
> I have no idea about your problem, I meet it in the training process, and training loss is decreasing normall, but get 0 map in the validation, it just appeared aside one of my custom cfgs, I guess I used wrong learning rate strategy, so I keep the cfg same as the original, and it goes well.

Hello DCC-Izhy,
Did you test the model, with the decreasing loss and map=0 is the model working as expected?
I did not use custom cfg, and still, I am facing the issue.
